src/puppetlabs/puppetdb/scf/storage.clj

```diff
@@ -1201,6 +1243,21 @@
   {:pre [(kitchensink/datetime? time)]}
   (jdbc/delete! :reports ["producer_timestamp < ?" (to-timestamp time)]))
 
+(defn delete-old-unassociated-catalogs!
+  [time]
+  {:pre [(kitchensink/datetime? time)]}
+  (when @store-catalogs-historically?
+    (jdbc/delete! :catalogs [(str "catalogs.producer_timestamp < ?"
+                                  "AND NOT EXISTS (SELECT 1 FROM"
+                                  "                (SELECT catalog_uuid FROM"
```

Comment:
This as well...  seems like 'not exists' shouldn't care about what you select, only that there are rows or not.